### PR TITLE
Fix item duplication and reduce debug message of putting things into wrong pocket

### LINKED ISSUE #41776: Certain Items Being Cloned On Load
# Describe the bug
When I load my save, the game displays an error (see screenshot), and several items (a wooden spear, a hacksaw, and a stone shovel) fall on the ground at my feet. These are all items that I _have_ had in my inventory at one point or another, but do not currently have in my inventory when saving the game. In the case of the hacksaw, the original doesn't even exist, as it has been crafted into a survivor utility belt.

# Steps To Reproduce
1. Load the attached save.
2. Move one square.

# Expected behavior
For items to not spontaneously appear on the ground at my feet when I load the game.

# Screenshots
The Error

The Items (multiple due to testing, it's 1 of each per load)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-3669-g1c69950 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi]
]

# Additional context
[save_and_log.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4866632/save_and_log.zip)

### PR DESCRIPTION
**Pull request by Menschheit**
#### Summary

SUMMARY: Bugfixes "Fix item duplication and fix losing empty pockets after loading"

#### Purpose of change

1. Trying to fix #41776.

2. Make sure empty pocket will be serialized so that the game can find the right pocket for items after loading.

#### Describe the solution

In #41484 I forgot to delete some codes used for testing (Apology, I wanted to make the game playable as soon as possible, but I am not experienced in coding). 
I deleted them, and modified `item_contents::combine` such as `item_pocket::insert_item` a little bit to further prevent duplication

**_The original codes will not serialize pocket to json if it is empty (similiar thing has caused mods disappearing), that will break the 1-1-correspondence, resulting trying to put things into wrong pocket, I removed the if-condition to let emtpy pocket to be saved as well._**

#### Describe alternatives you've considered

Just delete my codes and leave the others.

#### Testing

Using the save provided by  #41776. Doing this: load =>move 1 tile => save =>load =>move 1 tile=>....
Before my change, every time I load the game and move, duplication occurs.
After my change, only the first time there are items dropping to the floor(they possibly come from MOD pocket ) 

#### Additional context

Still need more testing but I can't do it my self.

### CODE REVIEW

File: src/item_pocket.cpp [hunk 1]
```diff
@@ -1299,6 +1299,7 @@ ret_val<item_pocket::contain_code> item_pocket::insert_item( const item &it )
     const ret_val<item_pocket::contain_code> ret = can_contain( it );
     if( contain_override || ret.success() ) {
         contents.push_back( it );
+        ret.make_success(); //may be bad for debugging but better than possible dublication
```
Comment (BevapDin) [inline src/item_pocket.cpp:1302]: Did you miss a `return` here? Currently this has no effect. `make_success` is a static function.

**Comment by Menschheit** [comment]
Sorry sometimes I only have a unclear image how things work before doing something. Is it correct now?

**Comment by Menschheit** [comment]
Changed the condition when a mod should go into MOD pocket.
If rail-mounted crossbow is put into MIGRATION pocket of the gun, the game crashes when I examine that gun, dunno why.

**Comment by misterprimus** [comment]
You can try to test with this: https://github.com/CleverRaven/Cataclysm-DDA/files/4871097/save.zip

**Comment by Menschheit** [comment]
> You can try to test with this: https://github.com/CleverRaven/Cataclysm-DDA/files/4871097/save.zip

Well I think it works. There are four two-way-radios in your Inventory, one of them is hidden(in MOD pocket), every time after loading, only one of them will drop to the ground, but you won't have more than four.

Without the change, every time after loading, a **new** radio will pop out and  they will duplicate infinity after saving/loading

**Comment by misterprimus** [comment]
> > You can try to test with this: https://github.com/CleverRaven/Cataclysm-DDA/files/4871097/save.zip
> 
> Well I think it works. There are four two-way-radios in your Inventory, one of them is hidden(in MOD pocket), every time after loading, only one of them will drop to the ground, but you won't have more than four.
> 
> Without the change, every time after loading, a **new** radio will pop out and they can duplicate infinitely after saving/loading
> 
> I need to figure out why radio keep poping out from army pants even if it can be contained.
> _**Edit: The original codes will not serialize pocket to json if it is empty (similiar thing has caused mods disappearing), that will break the 1-1-correspondence, resulting trying to put things into wrong pocket. After that change, you should only be able to see the debug message once, after re-saving and loading everything should be fine.**_

Sounds good. I just wanted to check:
1. Before applying your PR changes, were you reliably getting about 2 errors every time you loaded the game?
2. After your changes, did you stop getting the error on load?

**Comment by misterprimus** [comment]
Seems like this is a save/load issue? I was always wondering: why doesn't CDDA use the database approach like the old Aurora4X? I think the way it works is it basically keeps all the data in a database file. So if you quit, there's really nothing to "save," it just closes out of the file. If you want to back up the "save," you just make a copy of the file. That seems to prevent a whole bunch of issues like this since the devs don't have to worry about programming a save/load feature as that's done inherently by the way the gamestate is maintained during play.

**Comment by Menschheit** [comment]
1. I have only one error after loading
2. The first time you load the game the error occurs, save and load again the errors should be then eliminated.

**Comment by Menschheit** [comment]
> Seems like this is a save/load issue? I was always wondering: why doesn't CDDA use the database approach like the old Aurora4X? I think the way it works is it basically keeps all the data in a database file. So if you quit, there's really nothing to "save," it just closes out of the file. If you want to back up the "save," you just make a copy of the file. That seems to prevent a whole bunch of issues like this since the devs don't have to worry about programming a save/load feature as that's done inherently by the way the gamestate is maintained during play.

I don't know the game you mentioned, haven't played many old-school games.  Spend too much time playing trash games like DOTA2 cuz all my friends(well "all" means 4 or 5, sad story) like to play that.

For your question:
This is experimental build and backward compatibility is hard. So if the savefile has conflicts with the current game, either fix it with some bugs or just refuse to load it. The latter is always bad since you don't want to lose your save just because of one json change.

**Comment by misterprimus** [comment]
Well hopefully this fixes it and a dev can confirm it doesn't reintroduce a problem this code was initially meant to solve.

**Comment by Menschheit** [comment]
Close this since most of the changes are covered by #41917

**Comment by misterprimus** [comment]
Is this supposed to fix https://github.com/CleverRaven/Cataclysm-DDA/issues/41776 ?

**Pull request closed without merging**